### PR TITLE
Modal support within Nav Guards

### PR DESCRIPTION
**Pull request by will0684**
- In previous configuration modals would interrupt navigation tracking
- Android back button override now checks for if a modal is open, and then dismisses if so

### CODE REVIEW

File: src/pages/questionaire/questionaire.ts [hunk 1]
```diff
@@ -52,25 +53,41 @@ export class QuestionairePage {
...
   // Navbar Back button override
-  setBackButtonAction() {
+  setNavbarBackButtonAction() {
     this.navBar.backButtonClick = () => {
       this.contextCheck();
       this.navCtrl.pop().catch((reason: any) => this.logger.error(reason));
     };
   }
+  // Register override for Android hardware back button
+  setAndroidBackButtonAction() {
```
Comment (CalvinRodo) [inline src/pages/questionaire/questionaire.ts]: This method needs a test

File: src/pages/questionaire/questionaire.ts [hunk 2]
```diff
@@ -52,25 +53,41 @@ export class QuestionairePage {
...
-    this.platform.registerBackButtonAction(() => {
-      this.contextCheck();
-      this.navCtrl.pop().catch((reason: any) => this.logger.error(reason));
-    });
+    this.setNavbarBackButtonAction();
+    this.setAndroidBackButtonAction();
   }
   // Navbar Back button override
-  setBackButtonAction() {
+  setNavbarBackButtonAction() {
```
Comment (CalvinRodo) [inline src/pages/questionaire/questionaire.ts]: This method needs a test

File: src/pages/submission/submission.ts [hunk 3]
```diff
@@ -59,20 +61,36 @@ export class SubmissionPage {
...
   }
   // Navbar Back button override
-  setBackButtonAction() {
+  setNavbarBackButtonAction() {
     this.navBar.backButtonClick = () => {
       this.contextCheck();
       this.navCtrl.pop().catch((reason: any) => this.logger.error(reason));
     };
   }
+  setAndroidBackButtonAction() {
```
Comment (CalvinRodo) [inline src/pages/submission/submission.ts]: This method needs a test

**Comment by will0684** [comment]
I've pushed a new commit with very basic testing. I am going to continue building on my understanding of unit tests and solving this issue but any insight into the errors that Karma is sending back currently would be appreciated. @CalvinRodo @marjan-tropper

**Pull request merged**